atlas/AtlasMl/tests/test_pipelines_unit.py

```
@@ -0,0 +1,106 @@
+        [   # clusters after clustering
+            {"vector": {"default": [0.8, 0.2]}, "properties": {"cluster_id": "0"}}
+        ]
+    ])
+    workflows.weaviate_client.add_embeddings = MagicMock()
+    workflows.weaviate_client.update_property_by_id = MagicMock()
+    workflows.weaviate_client.get_embeddings_by_property = MagicMock(return_value=[
+        {"properties": {"competency_id": "K1"}}
+    ])
+    with patch("atlasml.ml.MLPipelines.PipelineWorkflows.apply_hdbscan") as mock_hdbscan, \
+         patch("atlasml.ml.MLPipelines.PipelineWorkflows.compute_cosine_similarity") as mock_cosine:
+        mock_hdbscan.return_value = ([0, 1], np.array([[0.8, 0.2]]), np.array([[0.8, 0.2]]))
```

Ruff (0.11.9)

Line too long (92 > 88)

(E501)